include/knowhere/index.h

```diff
@@ -16,6 +16,11 @@
 #include "knowhere/index_node.h"
 #include "knowhere/log.h"
 
+#ifdef NOT_COMPILE_FOR_SWIG
```

Comment:
Didn't see it defined/used/documented anywhere in this PR.

Reply:
This MACRO "NOT_COMPILE_FOR_SWIG" is defined in knowhere/CMakeLists.txt Line 62
The code embraced by this MACRO will not be compiled by python wheel.
Using this MACRO here is because python wheel need not compile prometheus metrics.